src/main/java/org/prebid/server/bidder/yieldmo/YieldmoBidder.java

```diff
@@ -1,52 +1,130 @@
 import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
 
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
 import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
 
-public class YieldmoBidder extends OpenrtbBidder<ExtImpYieldmo> {
+public class YieldmoBidder implements Bidder<BidRequest> {
+
+    private static final JsonPointer EXT_DATA_PBADSLOT_POINTER_EXPRESSION = JsonPointer.valueOf("/data/pbadslot");
+    private static final TypeReference<ExtPrebid<?, ExtImpYieldmo>> YIELDMO_EXT_TYPE_REFERENCE =
+            new TypeReference<>() {
+            };
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
 
     public YieldmoBidder(String endpointUrl, JacksonMapper mapper) {
-        super(endpointUrl, RequestCreationStrategy.SINGLE_REQUEST, ExtImpYieldmo.class, mapper);
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
     }
 
     @Override
-    protected Imp modifyImp(Imp imp, ExtImpYieldmo impExt) throws PreBidException {
-        final Imp.ImpBuilder modifiedImp = imp.toBuilder();
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest bidRequest) {
+        final List<Imp> modifiedImps = new ArrayList<>();
+        final List<BidderError> errors = new ArrayList<>();
+
+        for (Imp imp : bidRequest.getImp()) {
+            try {
+                final ExtImpYieldmo impExt = parseImpExt(imp);
+                modifiedImps.add(modifyImp(imp, impExt));
+            } catch (PreBidException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
+            }
+        }
 
-        final JsonNode pbadslotNode = imp.getExt().at("/data/pbadslot");
-        final String gpid = !pbadslotNode.isMissingNode()
-                ? StringUtils.defaultIfEmpty(pbadslotNode.asText(), null)
-                : null;
+        if (modifiedImps.isEmpty()) {
+            return Result.withErrors(errors);
+        }
+
+        final BidRequest modifiedRequest = bidRequest.toBuilder().imp(modifiedImps).build();
+        return Result.of(Collections.singletonList(makeRequest(modifiedRequest)), errors);
+    }
 
+    private ExtImpYieldmo parseImpExt(Imp imp) throws PreBidException {
         try {
-            modifiedImp.ext(mapper.mapper().valueToTree(YieldmoImpExt.of(impExt.getPlacementId(), gpid)));
+            return mapper.mapper().convertValue(imp.getExt(), YIELDMO_EXT_TYPE_REFERENCE).getBidder();
         } catch (IllegalArgumentException e) {
             throw new PreBidException(e.getMessage(), e);
         }
+    }
+
+    private Imp modifyImp(Imp imp, ExtImpYieldmo ext) {
+        final JsonNode pbadslotNode = imp.getExt().at(EXT_DATA_PBADSLOT_POINTER_EXPRESSION);
+        final String gpid = StringUtils.defaultIfBlank(pbadslotNode.asText(), null);
```

Comment:
For easier readability let's check that pbadslotNode is not MissingNode than use asText() method

Reply:
Do you wanna smth like
```
final String gpid = !pbadslotNode.isMissingNode()
                ? StringUtils.defaultIfBlank(pbadslotNode.asText(), null)
                : null;
```